modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -170,12 +171,17 @@ func (im IBCMiddleware) OnChanCloseConfirm(
 }
 
 // OnRecvPacket implements the IBCMiddleware interface
-func (IBCMiddleware) OnRecvPacket(
+func (im IBCMiddleware) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 	_ sdk.AccAddress,
 ) ibcexported.Acknowledgement {
 	err := errorsmod.Wrapf(icatypes.ErrInvalidChannelFlow, "cannot receive packet on controller chain")
+
+	if err != nil {
```

Comment:
err will always be non-nil here?

Reply:
we create the err on line 179, the check to see if it isn't nil is not needed is what I'm trying to say :smile:

Reply:
I know this PR is not needed but I added a comment.
```It would be good to add special message like 27-interchain-accounts: recv packet handling error so that it can be easy to check the log.
On ICA product development, debugging the failure takes a lot of time, and this simple log change would reduce a lot of time debugging.```

Reply:
```It would be good to add special message like 27-interchain-accounts: recv packet handling error so that it can be easy to check the log.```

Reply:
```On ICA product development, debugging the failure takes a lot of time, and this simple log change would reduce a lot of time debugging.```

Reply:
You want me to close this PR?

Reply:
sorry, wasn't clear! the point I was trying to make is that `if err != nil` is superfluous here, i.e, it could just be:

```
im.keeper.Logger(ctx).Error(fmt.Sprintf("27-interchain-accounts: recv packet handling error: %s", err.Error()))
```

without guarding it with the `if`.

Reply:
But damiannolan said it's already implemented in [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/host/ibc_module.go#L128)

Reply:
> It would be good to add special message like 27-interchain-accounts: recv packet handling error so that it can be easy to check the log.

You should be able to grep the logs pretty easily for `27-interchain-accounts` logs.

The logger used includes this within its context. See here -> https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/host/keeper/keeper.go#L85. 

Look for: `"module": "x/ibc-interchainaccounts"`